File: poo_py/car.py
class Car:
    def __init__(self):
        self.psg=0
        self.psgMax=2
        self.fuel=0
        self.fuelMax=100
        self.km=0
    
    def enter(self):
        if self.psg >= self.psgMax:
            print("max of passengers reached")
        else:
            self.psg+=1
        
    def leave(self):
        if self.psg == 0:
            print("empty car")
        else:
            self.psg-=1
    
    def drive(self, value):
        if self.psg == 0:
            print("empty car")
        elif self.fuel == 0:
            print("no fuel in the car")
        elif self.fuel < value:
            print(f"empty tank after ride {self.fuel} km")
            self.km+=self.fuel
            self.fuel=0
        else:
            self.fuel-=value
            self.km+=value

    def toFuel(self, value):
        self.fuel+=value
        if self.fuel > self.fuelMax:
            self.fuel = self.fuelMax % self.fuel

    def toString(self):
        print(f"pass:{self.psg}; gas:{self.fuel}; km:{self.km}")
        
def chain():
    car = Car()
    print("Commands:\n-end\n-show\n-enter\n-leave\n-fuel\n-drive")

    while 1:
        cmd=input("Insert a command:")
        if cmd == "end":
            break
        elif cmd == "show":
            car.toString()
        elif cmd == "enter":
            car.enter()
        elif cmd == "leave":
            car.leave()
        elif cmd == "fuel":
            value=int(input("Insert the value of fuel:"))
            car.toFuel(value)
        elif cmd == "drive":
            value=int(input("Insert the value of km:"))
            car.drive(value)
        else:
            print("Command not found...")
            
chain()

File: poo_cpp/car.cpp
#include <iostream>
using namespace std;

class Car {
//atributes
int pass=0;
int passMax=2;
int gas=0;
int gasMax=100;
int km=0;

//methods
public:

Car(){}

void enter(){
    if(this->pass >= 2)
        cout << "Full car!\n";
    else
        this->pass+=1;
}

void leave(){
    if(this->pass <= 0)
        cout << "Empty car!\n";
    else
        this->pass-=1;
}

void drive(int value){
    if(this->pass <= 0)
        cout << "Empty car!\n";
    else if(this->gas <= 0)
        cout << "Empty tank\n";
    else if(value > this->gas){
        cout << "Empty tank after ride " << this->gas << " km\n";
        this->km+=gas;
        gas=0;
    } else {
        this->gas-=value;
        this->km+=value;
    }
}

void toFuel(int value){
    this->gas+=value;
    if(this->gas > this->gasMax)
        this->gas=this->gasMax%gas;

}

void show(){
    cout <<"pass:" << this->pass 
    <<", gas: " << this->gas << ", km:" 
    << this->km << "\n";
}

};

void commandLine(Car car){
string cmd;
cout << "Type \"help\" to show all commands.\n";

while(true){
cout << "Insert a command: ";
cin >> cmd;

if(cmd == "end"){
    break;
} else if(cmd == "show"){
    car.show();
} else if(cmd == "enter"){
    car.enter();
} else if(cmd == "leave"){
    car.leave();
} else if(cmd == "fuel"){
    int value=0;
    cout << "Insert the value of fuel: ";
    cin >> value;
    car.toFuel(value);

} else if(cmd == "drive"){
    int value=0;
    cout << "Insert the value of km: ";
    cin >> value;
    car.drive(value);
} else if(cmd == "help"){
    cout << "Commands:\n-end\n-show\n-enter\n-leave\n-fuel\n-drive-\n";
} else {
    cout << "Command not found...\n";
}
}

}

int main() {

Car car;
commandLine(car);

}
